pandas/tests/groupby/test_categorical.py

```diff
@@ -860,3 +860,15 @@ def test_groupby_multiindex_categorical_datetime():
     expected = pd.DataFrame(
         {'values': [0, 4, 8, 3, 4, 5, 6, np.nan, 2]}, index=idx)
     assert_frame_equal(result, expected)
+
+
+def test_groupby_agg_observed_true_single_column():
+    # GH-23970
+    expected = pd.DataFrame({
```

Comment:
`df` and `expected` should be constructed as separate objects. I think generally this test case could be made more explicit depending on answer to other questions, but once we can clarify those pieces will want to split this out as two separate variables

Reply:
I can create a test case where `df` and `expected` are different but that'll cause `df` to be larger. 

I originally had `expected` and `df` as separate variable names but on @mroeschke 's advice (in a previous resolved convo) I replaced `df` with `expected`.

Reply:
can you parameterize over as_index=True/False

Reply:
I was parameterising as_index, but I hit a roadblock - I'm having difficulty recreating the expected value for as_index=True and I'm stumped how to go about fixing it.
Any help is greatly appreciated.
```
import pandas as pd
from pandas.util import testing as tm

expected = pd.DataFrame({
    'a': pd.Series([1, 1, 2], dtype='category'),
    'b': [1, 2, 2],
    'x': [1, 2, 3]
})

result = expected.groupby(['a', 'b'], 
	as_index=True,
	observed=True)['x'].sum()

print(result)

"""
a  b
1  1    1
   2    2
2  2    3
Name: x, dtype: int64
"""

s_idx = pd.MultiIndex(levels=[[1, 2], [1, 2]],
           codes=[[0, 0, 1], [0, 1, 1]],
           names=['a', 'b'])

s_val = [ 1, 2, 3 ]

expected = pd.Series(index=s_idx, data=s_val, name='x')
print(expected)

"""
a  b
1  1    1
   2    2
2  2    3
Name: x, dtype: int64
"""

tm.assert_series_equal(result, expected)

"""
Traceback (most recent call last):
  File "myTest.py", line 41, in <module>
    tm.assert_series_equal(result, expected)
  File "/home/vagrant/pandas/pandas/util/testing.py", line 1373, in assert_series_equal
    obj='{obj}.index'.format(obj=obj))
  File "/home/vagrant/pandas/pandas/util/testing.py", line 942, in assert_index_equal
    check_exact=check_exact, obj=lobj)
  File "/home/vagrant/pandas/pandas/util/testing.py", line 915, in assert_index_equal
    _check_types(left, right, obj=obj)
  File "/home/vagrant/pandas/pandas/util/testing.py", line 891, in _check_types
    assert_class_equal(l, r, exact=exact, obj=obj)
  File "/home/vagrant/pandas/pandas/util/testing.py", line 996, in assert_class_equal
    repr_class(right))
  File "/home/vagrant/pandas/pandas/util/testing.py", line 1188, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: MultiIndex level [0] are different

MultiIndex level [0] classes are not equivalent
[left]:  CategoricalIndex([1, 1, 2], categories=[1, 2], ordered=False, name='a', dtype='category')
[right]: Int64Index([1, 1, 2], dtype='int64', name='a')
"""

```
I understand that the issue is with not using CategoricalIndex at MultiIndex[0] but I have no idea how to go about doing it.

Reply:
use ``MultiIndex.from_arrays`` and pass a list of the Index already of the correct types